express-notes-api/index.js

```diff
@@ -58,12 +57,15 @@ app.delete('/api/notes/:id', (req, res) => {
     res.status(404).send(message404);
   } else if (dataJSON.notes[id]) {
     delete dataJSON.notes[id];
-    res.sendStatus(204);
     const newObj = JSON.stringify(dataJSON, null, 2);
     fs.writeFile('./data.json', newObj, err => {
       const message500 = { error: 'An unexpected error occured.' };
-      if (err) console.error(res.status(500).send(message500));
+      if (err) {
+        console.error(err);
+        res.status(500).send(message500);
+      }
     });
+    res.sendStatus(204);
```

Comment:
Your success responses need to move into your writeFile callback. We don't know when exactly writeFile will finished, except when the callback is called. So you need an `else` in each writeFile callback to deal with the success case.